Jropp/jason-ropp-js-iobp
# Onboarding Project: User component refactor

## What It Does

Simplifies, reorganizes, and sometimes renames the code to make it more maintainable.

As the lines of code has grown, depreciated functions, odd function order, and obsolete function naming has come up like so many little weeds. To deal with this I:

- I rearranged functions to be grouped by what they do. `formWatcher()` for example is followed by `updateDisableSave()` which `formWatcher()` uses.

- Renamed some of the functions to better reflect what they do. `fieldsFilled()` was replaced by a question `areFieldsFilled()` which more intuitively reflects that it returns a yes/no.

- Deleted functions that were no longer being used or were redundant.

- I created an `this.isClass(this.classes.CLASS_NAME)` function to replace all the `if()` and `ternary` instances of (this.mode === this.classes.CLASS_NAME). As the replacement made it more human readable. (The function will be used for dom-ifs in future PRs as well.);

No functionality was changed in this process.

## How To Test

- As you review the changes, check for the readability of the code. Do the function name changes appropriately and intuitively reflect what they do?

- Look at the overall layout of the functions. Does the order make sense? If you dig into a single function are the callbacks it uses easy to get to in the code order? Does that order and naming aid understanding of what is going on in the function?

- Are there any functions where the internal code seems jumbled or unclear? 

## Notes/Caveats

## Checklist

Under penalty of public shaming, I avow that I:


Browsers tested:

## Dependencies

- [ ] #98
